src/2ls/summary_checker_nonterm.cpp

```diff
@@ -0,0 +1,566 @@
+            init_expr=if_expr.false_case();
+            lb_var=if_expr.true_case();
+            //should already be renamed for inner loops
+            break;
+          }
+        }
+
+        loopback_vars.push_back(lb_var);
+        loopback_vars.push_back(init_expr);
+        loopback_vars.push_back(constk);
+      }
+
+      if (first_linearity_check.empty())  //nothing to be checked
+        continue;
+
+      neg_loop_exit_cond.push_back(lhguard);
+      neg_loop_exit_cond.push_back(not_exprt(lhguard_second));
+      loop_exit_cond.push_back(lhguard);
+
+      solver.new_context();
+      solver << conjunction(first_linearity_check);
+      switch(solver())
+      {
+      case decision_proceduret::D_UNSATISFIABLE: //
+          //std::cout << "£Unsat++++++++++++++++" << std::endl;
+          solver.pop_context();
+        continue;
+
+      case decision_proceduret::D_SATISFIABLE:
+          //std::cout << "£Sat++++++++++++++++" << std::endl;
+          for (exprt::operandst::iterator it=first_linearity_check.begin();
+               it!=first_linearity_check.end();
+               ++it)
+          {
+            exprt ex=solver.get(it->op1().op1());
+            second_linearity_check.push_back(
+                  and_exprt(*it, not_exprt(equal_exprt(to_constant_expr(ex),
+                                                       it->op1().op1()))));
+          }
+
+          solver.pop_context();
+        break;
+
+      default:
+        error() << "decision procedure has failed" << eom;
+        solver.pop_context();
+        solver.pop_context();
+        return;
+      }
+
+      solver.new_context();
+      solver << disjunction(second_linearity_check);
+      switch(solver())
+      {
+      case decision_proceduret::D_UNSATISFIABLE:
+          //std::cout << "£Unsat++++++++++++++++" << std::endl;
+          solver.pop_context();
+        break;
+
+      case decision_proceduret::D_SATISFIABLE:
+          //std::cout << "£Sat++++++++++++++++" << std::endl;
+          solver.pop_context();
+        continue;
+
+      default:
+        error() << "decision procedure has failed" << eom;
+        solver.pop_context();
+        solver.pop_context();
+        return;
+      }
+
+      // constants extraction
+      exprt::operandst solver_consts;
+
+      solver.new_context();
+      solver << conjunction(constants_computation);
+      switch(solver())
+      {
+      case decision_proceduret::D_UNSATISFIABLE: // should never happen
+          //std::cout << "£Unsat++++++++++++++++" << std::endl;
+          solver.pop_context();
+          solver.pop_context();
+        return;
+
+      case decision_proceduret::D_SATISFIABLE:
+          //std::cout << "£Sat++++++++++++++++" << std::endl;
+          for (auto constant : constants)
+          {
+            exprt ex = solver.solver->get(constant);
+            if (ex.type().id()==ID_unsignedbv)
+            {
+              // if (constant>UINT_MAX/2)?0-constant:constant
```

Comment:
Does not matter whether I go minus or plus direction. -1 is the same as +UINT_MAX and I always cover the path. However for UINT_MAX the algorithm is very uneffective since I terate the range of unsigned int while for 1 I will cover the whole path at once. This transformation replaces high numbers with their complement.